asynciterator.ts

```diff
@@ -25,6 +25,11 @@ export function setTaskScheduler(scheduler: TaskScheduler): void {
   taskScheduler = scheduler;
 }
 
+/** Binds a function to an object */
+function bind(fn: Function, self?: object) {
```

Comment:
```suggestion
function bind<T extends Function>(fn: T, self?: object): T {
```

is the correct one, but that leads to errors below.